gen_code_can_common.c

```diff
@@ -0,0 +1,820 @@
+    
+        output->data[3] = degs; 
+    output->data[4] = mins; 
+    output->data[5] = dmins; 
+
+    
+    output->data_len = 7;
+
+    return true;
+}
+        
+bool build_gps_longitude_msg(uint32_t timestamp,
+                                                   uint8_t degs, 
+                           uint8_t mins, 
+                           uint16_t dmins, 
+
+                        can_msg_t *output)
+
+{
+    if (!output) { return false; }
+
+    output->sid = MSG_GPS_ALTITUDE | BOARD_UNIQUE_ID;
+    write_timestamp_3bytes(timestamp, output);        
+    
+        output->data[3] = degs; 
+    output->data[4] = mins; 
+    output->data[5] = dmins; 
+
+    
+    output->data_len = 7;
+
+    return true;
+}
+        
+bool build_gps_altitude_msg(uint32_t timestamp,
+                                                   uint16_t altitude, 
+                           uint8_t daltitude, 
+
+                        can_msg_t *output)
+
+{
+    if (!output) { return false; }
+
+    output->sid = MSG_GPS_ALTITUDE | BOARD_UNIQUE_ID;
+    write_timestamp_3bytes(timestamp, output);        
+    
+        output->data[3] = altitude; 
+    output->data[4] = daltitude; 
+
+    
+    output->data_len = 6;
+
+    return true;
+}
+        
+bool build_gps_info_msg(uint32_t timestamp,
+                                                   uint8_t num_sats, 
+                           uint8_t quality, 
+
+                        can_msg_t *output)
+
+{
+    if (!output) { return false; }
+
+    output->sid = MSG_GPS_ALTITUDE | BOARD_UNIQUE_ID;
+    write_timestamp_3bytes(timestamp, output);        
+    
+        output->data[3] = num_sats; 
+    output->data[4] = quality; 
+
+    
+    output->data_len = 5;
+
+    return true;
+}
+        
+bool build_fill_lvl_msg(uint32_t timestamp,
+                                                   uint8_t level, 
+       enum FILL_DIRECTION fill_direction, 
+
+                        can_msg_t *output)
+
+{
+    if (!output) { return false; }
+
+    output->sid = MSG_GPS_ALTITUDE | BOARD_UNIQUE_ID;
+    write_timestamp_3bytes(timestamp, output);        
+    
+        output->data[3] = level; 
+    output->data[4] = (uint8_t) fill_direction; 
+
+    
+    output->data_len = 5;
+
+    return true;
+}
+        
+bool build_radi_value_msg(uint32_t timestamp,
+                                                   uint8_t radi_board, 
+                           uint16_t radi, 
+
+                        can_msg_t *output)
+
+{
+    if (!output) { return false; }
+
+    output->sid = MSG_GPS_ALTITUDE | BOARD_UNIQUE_ID;
+    write_timestamp_3bytes(timestamp, output);        
+    
+        output->data[3] = radi_board; 
+    output->data[4] = radi; 
+
+    
+    output->data_len = 5;
+
+    return true;
+}
+        
+int get_general_cmd(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    
+
+}
+
+int get_actuator_cmd(const can_msg_t *msg,
```

Comment:
name: get_actuator_id
missing:
`
        uint16_t msg_type = get_message_type(msg);
        switch (msg_type) {
        case MSG_ACTUATOR_CMD:
        case MSG_ACTUATOR_STATUS:
        return msg->data[3];
        default:
            // not a valid field for this message type
            return -1;
    }
 `